src/components/lib/histogram.js

```diff
@@ -15,12 +15,55 @@ function getXScale(data, width) {
     nice();
 }
 
+// Computes the average of an array of numbers. If the array is empty, returns 1.
+function avg(arr) {
+  return arr.length > 0 ? arr.reduce((a,b)=>a+b)/arr.length : 1
+}
+
+// Computes min(|a|,|b|)/max(|a|,|b|).
+function fractionLT1(a,b) {
+  return Math.min(Math.abs(a),Math.abs(b))/Math.max(Math.abs(a),Math.abs(b))
+}
+
+function filterData(inputData, cutOff) {
+  // We can't filter that intelligently for small sample sets, so we don't bother.
+  if (inputData.length < 2000) {
+    return inputData
+  }
+
+  let outputData = inputData // A copy for immutability
```

Comment:
I'm not sure if this would actually do a deep clone.  The traditional way is to use ``Object.assign({}, inputData)`` or a lodash clone